relays/ethereum/src/sync_loop.rs

```diff
@@ -288,14 +290,24 @@ pub fn run<P: HeadersSyncPipeline>(
 						|| format!("Error retrieving existence status from {} node", P::TARGET_NAME),
 					);
 				},
-				target_submit_header_result = target_submit_header_future => {
-					target_client_is_online = process_future_result(
-						target_submit_header_result,
-						|submitted_headers| sync.headers_mut().headers_submitted(submitted_headers),
-						&mut target_go_offline_future,
-						|| async_std::task::sleep(CONNECTION_ERROR_DELAY),
-						|| format!("Error submitting headers to {} node", P::TARGET_NAME),
-					);
+				submitted_headers = target_submit_header_future => {
+					// this closure helps Rust understand the type of `submitted_headers` :/
```

Comment:
Can't you do:
```rust
let submitted_headers: SubmittedHeaders<HeaderId<P::Hash, P::Number>, TC::Error> = submitted_headers;
```
instead of introducing a closure?

Reply:
yeah, thanks - this also helps :)